src/services/textChanges.ts

```diff
@@ -889,7 +889,7 @@ namespace ts.textChanges {
                 : format(change.node);
             // strip initial indentation (spaces or tabs) if text will be inserted in the middle of the line
             const noIndent = (options.preserveLeadingWhitespace || options.indentation !== undefined || getLineStartPositionForPosition(pos, sourceFile) === pos) ? text : text.replace(/^\s+/, "");
-            return (options.prefix || "") + noIndent + (options.suffix || "");
+            return (options.prefix || "") + noIndent + (((options.suffix && noIndent.match(options.suffix+"$")) || !options.suffix) ? "" : options.suffix);

```

Comment:
I don't like the implicit coercion of a string to a regexp without quoting, but it seems that this only checks for a suffix, so maybe something like the following?  Also, a single line is very hard to read, the following is a reorg + some simplification of the condition.

```suggestion
            return (options.prefix || "") + noIndent
                 + ((!options.suffix || endsWith(noIndent, options.suffix))
                    ? "" : options.suffix);
```

Reply:
modified

Reply:
Do you mind also indenting the `+ ...` part so it's not on the same level as the `return`?